app/src/interfaces/list-o2m/list-o2m.vue

```diff
@@ -423,7 +423,13 @@ function stageEdits(item: Record<string, any>) {
 	if (newItem) {
 		create(item);
 	} else {
-		update(item);
+		const updatePermission = permissionsStore.getPermissionsForUser(relatedCollection.value, 'update');
+		// prevent update payload containing fields that are included for displays, but without edit permission
+		const itemToUpdate =
+			updatePermission?.fields && !updatePermission.fields.includes('*')
+				? pick(item, [...updatePermission.fields, '$index', '$type', '$edits'])
```

Comment:
Not sure if there's a better way of handling this, or maybe something like a reduce and check for keys starting with `$`.

Reply:
Good idea. Lets make this a hair more generic by checking for all keys starting with `$` rather than an arbitrary hardcoded list 👍🏻